xdsl/transforms/stencil_to_csl_stencil.py

```diff
@@ -171,10 +295,314 @@ def match_and_rewrite(self, op: dmp.SwapOp, rewriter: PatternRewriter, /):
             nested_rewriter.rewrite_op(new_apply_op)
 
 
+def get_op_split(
+    ops: Sequence[Operation], prefetch: BlockArgument
+) -> tuple[Sequence[Operation], Sequence[Operation]]:
+    """
+    Returns a split of ops into an `(a,b)` tuple, such that:
+
+    - `a` contains neighbour accesses plus the minumum set of instructions to reduce the accessed data to 1 thing
+    - `b` contains everything else
+
+    If no valid split can be found, return `(ops, [])`.
+
+    This function does not attempt to arithmetically re-structure the computation to obtain a good split. To do this,
+    `RestructureSymmetricReductionPattern()` may be executed first.
+    """
+    a: Sequence[Operation] = []
+    b: Sequence[Operation] = []
```

Comment:
I found these one-letter names to become confusing later down the line when used in list comprehensions. Would you mind giving these more descriptive names (e.g. `dependent_ops` and `independent_ops`?)

Reply:
Splitting a list into two parts a and b is probably one of the few cases where I'd use single-char variable names, but I have renamed things in the list comprehensions hoping to make them more readable. Come back on this please if you don't agree.

Reply:
The thing is that `a` and `b` have semantic meaning attached, so when there's a check like `x in a` it's less clear than `x in independent_ops`.

Reply:
The semantic meaning that comes to mind is to refer to elements of a tuple 😅